### PR TITLE
s3: Add ArvanCloud AOS to provider list

### PR DESCRIPTION
**Pull request by ehsantdy**
#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

**Comment by ehsantdy** [comment]
ArvanCloud Object Storage (AOS)) is a cloud object storage service, and is fully compatible with S3.

**Comment by ehsantdy** [comment]
@ncw

File: backend/s3/s3.go [hunk 1]
```diff
@@ -2167,6 +2209,8 @@ func setQuirks(opt *Options) {
 		listObjectsV2 = false
 		virtualHostStyle = false
 		urlEncodeListings = false
+	case "ArvanCloud":
+		// No quirks
```
Comment (ncw) [inline backend/s3/s3.go:2213]: Did you check the integration tests pass to get the no quirks here?

File: docs/content/s3.md [hunk 2]
```diff
@@ -3541,6 +3579,114 @@ d) Delete this remote
 y/e/d> y
 ```
 
+### Wasabi
```
Comment (ncw) [inline docs/content/s3.md:3582]: This heading should say Arvan Cloud I think!

**Review by ncw — COMMENTED** [review]
This looks great!

I put a couple of notes inline for your attention.

Sorry for the delay reviewing

File: backend/s3/s3.go (hunk 1 above)
Comment (ncw) [reply backend/s3/s3.go]: If `go test -v -remote YourS3Remote:` passes OK with no quirks then I'm happy.

**Comment by ncw** [comment]
Does

    go test -v -remote YourS3RemotePointingToArvanCloud:

Pass OK?

**Comment by ncw** [comment]
I'll merge this as-is - if you want to send a followup PR adjusting anything then that is fine :-)

Thank you.

File: docs/content/s3.md [hunk 3]
```diff
@@ -1705,6 +1728,21 @@ The storage class to use when storing new objects in ChinaMobile.
 
 #### --s3-storage-class
 
+The storage class to use when storing new objects in ChinaMobile.
```
Comment (lrvl) [inline docs/content/s3.md:1731]: Notice, reused ChinaMobile

Comment (ehsantdy) [reply docs/content/s3.md:1731]: Thanks dear Leroy, i will correct it.

**Pull request merged**